client/data/users/use-users.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { useInfiniteQuery } from 'react-query';
+import { uniqueBy } from '@automattic/js-utils';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+export const defaults = {
+	number: 100,
+	order: 'ASC',
+	order_by: 'display_name',
+};
+
+const extractPages = ( pages = [] ) => pages.flatMap( ( page ) => page.users );
+const compareUnique = ( a, b ) => a.ID === b.ID;
+
+const useUsers = ( siteId, fetchOptions = {}, queryOptions = {} ) => {
```

Comment:
A naming idea: would `useUsersQuery` be a better name? It reveals more about what the hook actually does. It could be a good naming convention once we have dozens of `use*Query` hooks.

Reply:
Agreed. I actually wonder what led me to omitting the `Query` suffix 🤔

Reply:
Done via https://github.com/Automattic/wp-calypso/pull/50712/commits/6ade32e16df927ae6cd5565170b8ad477d9946ec

(I made note to add that convention to a README document which I plan on adding to the `client/data` directory)